tests/www/v1/stock2shop/dal/channels/memory/ChannelStateTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace tests\v1\stock2shop\dal\channels\memory;
+
+use stock2shop\dal\channels\memory;
+use tests;
+
+/**
+ * Channel State Test
+ */
+class ChannelStateTest extends tests\TestCase
+{
+
+    /**
+     * Test Get Products
+     * @return void
+     */
+    public function testGetProducts()
+    {
+        memory\ChannelState::clean();
+
+        $productCount = 11;
+        $newMemoryProducts = [];
+        for ($i = 0; $i < $productCount; $i++) {
+            $newMemoryProducts[] = new memory\MemoryProduct([
+                'id' => null,
+                'product_group_id' => 'cpid1',
+                'name' => 'Product Name',
+                'price' => '5000.00',
+                'quantity' => 5,
+                'images' => [
+                    'http://aws.stock2sho..1',
+                ]
+            ]);
+        }
+
+        memory\ChannelState::updateProducts($newMemoryProducts);
+        $this->assertCount(count($newMemoryProducts), memory\ChannelState::getProducts());
+
+        memory\ChannelState::clean();
+    }
+
+    /**
+     * Test Update Images
+     * @return void
+     */
+    public function testUpdateImages()
+    {
+        memory\ChannelState::clean();
+        $memoryImageOne = new memory\MemoryImage(['id' => null, 'url' => 'http://aws.stock2sho..1', 'product_group_id' => '1']);
+        $memoryImageTwo = new memory\MemoryImage(['id' => null, 'url' => 'http://aws.stock2sho..1', 'product_group_id' => '1']);
+        memory\ChannelState::updateImages([$memoryImageOne, $memoryImageTwo]);
+        $images = memory\ChannelState::getImages();
+        foreach ($images as $imageKey => $imageItem) {
+            $this->assertTrue($imageItem instanceof memory\MemoryImage);
+            $this->assertNotNull($imageItem->id);
+            $this->assertNotNull($imageItem->url);
+            $this->assertNotNull($imageItem->product_group_id);
+        }
+        memory\ChannelState::clean();
+    }
+
+    /**
+     * Test Generate ID
+     *
+     * Evaluates the method which generates a random string ID
+     * for use with images and products on the channel.
+     *
+     * @return void
+     */
+    public function testGenerateID()
+    {
+        memory\ChannelState::clean();
+        $outcome = memory\ChannelState::generateID();
+        $this->assertEquals('string', gettype($outcome));
+        $this->assertEquals(50, strlen($outcome));
+        memory\ChannelState::clean();
+    }
+
+    /**
+     * Test Get Images By Group IDs
+     *
+     * Evaluates that the method returns MemoryImage
+     * objects by "product_group_id".
+     *
+     * @return void
+     */
+    public function testGetImages()
+    {
+        memory\ChannelState::clean();
+        $groupIDs = ['cpid1', 'cpid2'];
+        $groupImages = [];
+        for ($i = 0; $i !== 2; $i++) {
```

Comment:
Why not use imageCount var the same as you have done in [testGetProducts](https://github.com/stock2shop/connector/pull/1/files?show-deleted-files=true&show-viewed-files=true&file-filters%5B%5D=#diff-6fab95f1a0843c4ce4d4405674398e1e7a4b43551ce78aa153e71c049b293ff9R24)

Reply:
I have added `$imageCount` and set the value to 2 and the variable is now used to determine the number of times to run the `for loop` which adds the images to the `memory\ChannelState`.